Extensions/combined/ryd.content-script.js

```diff
@@ -155,10 +156,16 @@ function RYD() {
   }
 
   function sendVote(vote) {
-    RYDTools.getBrowser().runtime.sendMessage({
-      message: "send_vote",
-      vote: vote,
-      videoId: getVideoId(window.location.href)
+    RYDTools.getBrowser().storage.sync.get(["do_submission"], (resp) => {
+      if(!resp["do_submission"]){
+        return;
+      }
+      console.log("Sending vote...");
+      RYDTools.getBrowser().runtime.sendMessage({
+        message: "send_vote",
+        vote: vote,
+        videoId: getVideoId(window.location.href)
+      });
```

Comment:
Why not make the if-clause check for the positive outcome and just let out the negative part all together.

```suggestion
    RYDTools.getBrowser().storage.sync.get(["do_submission"], (resp) => {
      if(resp["do_submission"]){
        console.log("Sending vote...");
        RYDTools.getBrowser().runtime.sendMessage({
          message: "send_vote",
          vote: vote,
          videoId: getVideoId(window.location.href)
        });
      }
```

Reply:
Also you should think about removing the console.log's. Once you start keeping those in for prod it soon starts to get out of hand imo